src/main/java/com/softserveinc/ita/pageobjects/admin/SubjectsPage.java

```diff
@@ -68,4 +59,20 @@ public AddingFormModal editSubject(String subject) {
     private void waitUntilModalVisible() {
         $x("//app-subjects-create-modal").shouldBe(visible);
     }
-}
+
+    @Step("Subjects page: Opened Tests page of {subject}")
+    public TestsPage openSubjectTests(String subject) {
+        setSearchValue(subject);
+        performActionWithSubject(subject,"assignment_turned_in");
+
+        return new TestsPage();
+    }
+
+    private void performActionWithSubject(String subject, String actionToPerform) {
```

Comment:
It looks similar to the delete method in EntutyTable, you could merge and move this method there

Reply:
just use "*" instead 'mat-icon' and 'i' at the same time